plugins/audit-log-node/src/DefaultAuditLogger.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  AuthService,
+  HttpAuthService,
+  LoggerService,
+} from '@backstage/backend-plugin-api';
+import { ErrorLike } from '@backstage/errors';
+
+import { Request } from 'express';
+
+import {
+  AuditErrorLogOptions,
+  AuditLogDetailsOptions,
+  AuditLogger,
+  AuditLoggerOptions,
+  AuditLogOptions,
+} from './types';
+
+export class DefaultAuditLogger implements AuditLogger {
+  private readonly logger: LoggerService;
+  private readonly authService: AuthService;
+  private readonly httpAuthService: HttpAuthService;
+
+  constructor(options: AuditLoggerOptions) {
+    this.logger = options.logger;
+    this.authService = options.authService;
+    this.httpAuthService = options.httpAuthService;
+  }
+
+  async getActorId(request?: Request): Promise<string | undefined> {
+    if (!(request && this.httpAuthService && this.authService)) {
+      return undefined;
+    }
+    try {
+      const credentials = await this.httpAuthService.credentials(request);
+      const userEntityRef = this.authService.isPrincipal(credentials, 'user')
+        ? credentials.principal.userEntityRef
+        : undefined;
+
+      const serviceEntityRef = this.authService.isPrincipal(
+        credentials,
+        'service',
+      )
+        ? credentials.principal.subject
+        : undefined;
+
+      return userEntityRef ?? serviceEntityRef;
+    } catch {
+      return undefined;
+    }
+  }
+  async createAuditLogDetails(options: AuditLogDetailsOptions) {
```

Comment:
Add return type, please.

Reply:
Then, I guess you will need to cast `auditLogDetails as JsonObject)` in the bottom of the code:

`this.logger.info(options.message, auditLogDetails as JsonObject);`

`this.logger.error(options.message, auditLogDetails as JsonObject);`

Reply:
Oh sorry, I forgot to port the return type in the original function over. It should be returning a `Promise<AuditLogDetails>` type.